UPDATING.md

```diff
@@ -0,0 +1,127 @@
+# Upgrade / Migration Guide
+
+## Version 1.1.5 to 1.2.0
+
+We have made many **breaking changes** in the version 1.2 release of this SDK.
+
+In this guide we aim to highlight the main differences you will encounter when migrating your code from the interfaces we were offering prior to the [version 1.2.0 release](https://github.com/ably/ably-go/releases/tag/v1.2.0).
+
+These include:
+
+- Changes to numerous function and method signatures - these are breaking changes, in that your code will need to change to use them in their new form
+- Adoption of the [Context Concurrency Pattern](https://blog.golang.org/context) using the [context package](https://pkg.go.dev/context) - e.g. [ably.RealtimeChannel.Subscribe](https://pkg.go.dev/github.com/ably/ably-go/ably#RealtimeChannel.Subscribe)
+- Adoption of the [Functional Options Pattern](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) - e.g. [ably.ClientOption](https://pkg.go.dev/github.com/ably/ably-go/ably#ClientOption)
+
+### Asynchronous Operations and the `Context` Concurrency Pattern
+
+We've now moved to a model where methods are blocking in nature, cancellable with a [Context](https://pkg.go.dev/context#Context).
+
+We use `ctx` in examples to refer to a `Context` instance that you create in your code.
+
+For robust, production-ready applications you will rarely (actually, probably _never_) want to create your `Context` using the basic [Background](https://pkg.go.dev/context#Background) function, because it cannot be cancelled and remains for the lifecycle of the program. Instead, you should use [WithTimeout](https://pkg.go.dev/context#WithTimeout) or [WithDeadline](https://pkg.go.dev/context#WithDeadline).
```

Comment:
Given my comment above, this is probably a bit misleading in the context of Realtime. The SDK has has its own timeout handling built in, as required by the spec and Realtime protocol.

Also, I'm not sure we should be explaining what contexts are for at all. They've been in common use for years, and are thoroughly documented elsewhere.

Reply:
That is a very good point and I understand what you are saying regarding timeouts. I'm going to see what words I can find to clarify this as it's worth highlighting.

I disagree, however, when it comes to the validity of explaining contexts. I tried to keep it as abstract as possible, but there will be developers from all levels approaching this documentation, so I believe this is a relevant expansion - especially given how core the adoption of this pattern is to the changes in 1.2.0.

Reply:
https://github.com/ably/ably-go/pull/364/commits/07f63ef1164ec4d20d4dc26dcb5fb878426f2ed2